lib/src/screens/connect_device/connect_device_page.dart

```diff
@@ -35,91 +35,104 @@ class ConnectDevicePage extends BasePage {
   String get title => S.current.restore_title_from_hardware_wallet;
 
   @override
-  Widget body(BuildContext context) => ConnectDevicePageBody(walletType, onConnectDevice, ledgerVM);
+  Widget body(BuildContext context) =>
+      ConnectDevicePageBody(walletType, onConnectDevice, ledgerVM);
 }
 
 class ConnectDevicePageBody extends StatefulWidget {
   final WalletType walletType;
   final OnConnectDevice onConnectDevice;
   final LedgerViewModel ledgerVM;
 
-  const ConnectDevicePageBody(this.walletType, this.onConnectDevice, this.ledgerVM);
+  const ConnectDevicePageBody(
+      this.walletType, this.onConnectDevice, this.ledgerVM);
 
   @override
   ConnectDevicePageBodyState createState() => ConnectDevicePageBodyState();
 }
 
 class ConnectDevicePageBodyState extends State<ConnectDevicePageBody> {
-  final imageLedger = 'assets/images/ledger_nano.png';
-
-  final ledger = Ledger(
-    options: LedgerOptions(
-      scanMode: ScanMode.balanced,
-      maxScanDuration: const Duration(minutes: 5),
-    ),
-    onPermissionRequest: (_) async {
-      Map<Permission, PermissionStatus> statuses = await [
-        Permission.bluetoothScan,
-        Permission.bluetoothConnect,
-        Permission.bluetoothAdvertise,
-      ].request();
-
-      return statuses.values.where((status) => status.isDenied).isEmpty;
-    },
-  );
-
-  var bleIsEnabled = true;
   var bleDevices = <LedgerDevice>[];
   var usbDevices = <LedgerDevice>[];
 
   late Timer? _usbRefreshTimer = null;
   late Timer? _bleRefreshTimer = null;
+  late Timer? _bleStateTimer = null;
   late StreamSubscription<LedgerDevice>? _bleRefresh = null;
 
   @override
   void initState() {
     super.initState();
     WidgetsBinding.instance.addPostFrameCallback((_) {
-      _bleRefreshTimer = Timer.periodic(Duration(seconds: 1), (_) => _refreshBleDevices());
+      _bleStateTimer = Timer.periodic(
+          Duration(seconds: 1), (_) => widget.ledgerVM.updateBleState());
```

Comment:
what does updateBleState do?
and what would happen if ble permission is not granted? would it throw an exception that we should catch here?